### PR TITLE
[Site Name] Adjusts design picker buttons for treatment variation

### PR DESCRIPTION
**Pull request by antonis**
## Description
This PR adjusts the design picker `Choose` buttons to communicate to the user that their action will create the site.

|Design Chooser(Control)|Design Chooser(Treatment)|
|---|---|
|IMG_0188|IMG_EB02383F75C5-1|

|Design Preview(Control)|Design Preview(Treatment)|
|---|---|
|IMG_0189|IMG_0187|

### Treatment variation
1. Toggle ON the Site Name feature
2. Start the Site Creation flow
3. Choose an intent for your site or skip
4. Enter a site name or skip
5. Select a design
6. Verify that the `choose` button is changed to `create site`
7. Preview a design and verify that the `choose` button is changed to `create site`

### Control variation
1. Toggle OFF the Site Name feature
2. On steps 6 and 7 above verify that the buttons are not changed

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by antonis** [comment]
Note: The PR will probably need to target `release/19.7`

**Comment by antonis** [comment]
The skip button change was reverted with 7a4701da8c06d2ee434688113414ac5a796e3098

File: WordPress/Classes/ViewRelated/Site Creation/Design Selection/Preview/TemplatePreviewViewController.swift
Comment (Gio2018) [inline WordPress/Classes/ViewRelated/Site Creation/Design Selection/Preview/TemplatePreviewViewController.swift]: since we have a couple of hardcoded strings, including this one, would it be possible to group them in a private `enum`?

File: WordPress/Classes/ViewRelated/Site Creation/Design Selection/Preview/TemplatePreviewViewController.swift
Comment (Gio2018) [inline WordPress/Classes/ViewRelated/Site Creation/Design Selection/Preview/TemplatePreviewViewController.swift]: nit: the comment should have a period at the end.

File: WordPress/Classes/ViewRelated/Site Creation/Design Selection/SiteDesignContentCollectionViewController.swift
Comment (Gio2018) [inline WordPress/Classes/ViewRelated/Site Creation/Design Selection/SiteDesignContentCollectionViewController.swift:65]: similar comments here: the comment should have a period, and since this file has quite a few hardcoded strings, wondering if we can group them?

**Review by Gio2018 — COMMENTED** [review]
Thanks for tackling this @antonis ! the change works as expected; just left a couple of comments for you to consider.

**Comment by twstokes** [comment]
👋 If we're going to land on a single string after the A/B tests I'd suggest not using this PR to keep things simpler, but I've created one with tests in case we might need it.
- https://github.com/wordpress-mobile/WordPress-iOS/pull/18400

**Review by twstokes — COMMENTED** [review]
My only suggestion would be to consider making changes in `SiteDesignPreviewViewController` rather than `TemplatePreviewViewController` since it's a more specific subclass. `LayoutPreviewViewController` also relies on `TemplatePreviewViewController` but because of ordering it isn't affected. 👍

File: WordPress/Classes/ViewRelated/Site Creation/Design Selection/Preview/TemplatePreviewViewController.swift
Comment (antonis) [reply WordPress/Classes/ViewRelated/Site Creation/Design Selection/Preview/TemplatePreviewViewController.swift]: Great suggestion @Gio2018 👍 
Handled with babc542445c63b38afa12643abe19746e3dc30cb

File: WordPress/Classes/ViewRelated/Site Creation/Design Selection/Preview/TemplatePreviewViewController.swift
Comment (antonis) [reply WordPress/Classes/ViewRelated/Site Creation/Design Selection/Preview/TemplatePreviewViewController.swift]: Handled with babc542445c63b38afa12643abe19746e3dc30cb 👍

File: WordPress/Classes/ViewRelated/Site Creation/Design Selection/SiteDesignContentCollectionViewController.swift
Comment (antonis) [reply WordPress/Classes/ViewRelated/Site Creation/Design Selection/SiteDesignContentCollectionViewController.swift:65]: As above handled with babc542445c63b38afa12643abe19746e3dc30cb. Thank you for the feedback 🙇

**Comment by antonis** [comment]
> 👋 If we're going to land on a single string after the A/B tests I'd suggest not using this PR to keep things simpler, but I've created one with tests in case we might need it.
> 
> * [[Site Creation] Pass a boolean to determine Site Design primary button titles #18400](https://github.com/wordpress-mobile/WordPress-iOS/pull/18400)

Thank you for your suggestions @twstokes 🙇 
I agree that your approach is a lot cleaner (can be unit tested 🎉 ) and might also be useful in our future enhancements (even when we remove the experiment). I've merged your draft PR onto this one :)

**Comment by antonis** [comment]
⚠️ Please check the response here (internal ref: `p5T066-37N-p2#comment-12065`) before merging to the frozen `19.7` release 🙇

**Review by ovitrif — APPROVED** [review]
All good during testing (both with Site Name on / off) and the way the "shape" of the code improved is quite nice! 🚀 

Thank you @antonis & @twstokes for your collaborative efforts on this, and @Gio2018 for the valuable feedback 🙇.

I'm approving the PR, but as per the **[PR merge policy](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/CONTRIBUTING.md#pr-merge-policy)** for our iOS repos, the author would normally merge it.

Apparently we're still in need of a follow-up to this internal comment: p5T066-37N-p2#comment-12066

**Review by kyleaparker — APPROVED** [review]

File: WordPress/Classes/ViewRelated/Site Creation/Design Selection/Preview/SiteDesignPreviewViewController.swift
Comment (twstokes) [inline WordPress/Classes/ViewRelated/Site Creation/Design Selection/Preview/SiteDesignPreviewViewController.swift:31]: Nice, good suggestion @Gio2018 and implementation @antonis. Makes things a lot cleaner!

**Review by twstokes — APPROVED** [review]
LGTM! 🙇 If I understand correctly about how the translation keys work, the interface should translate "Create Site" because [that key has been translated before](https://github.com/wordpress-mobile/WordPress-iOS/blob/7f348357038e29bdc16268d8f234962579eae59b/WordPress/Classes/ViewRelated/Site%20Creation/Web%20Address/WebAddressWizardContent.swift#L428) and the comment isn't part of the key.

**Review by Gio2018 — APPROVED** [review]
Awesome @antonis ! :shipit: !

**Comment by mokagio** [comment]
@antonis and @twstokes (writing a single comment in the main PR) this has been bundled as part of [19.7 beta 1 (19.7.0.1)](https://github.com/wordpress-mobile/WordPress-iOS/pull/18411).

Thanks for your work 🙌

**Pull request merged**
